New package: StitchBindings v0.1.0

JuliaRegistrator opened the pull request:
- Registering package: StitchBindings
- Repository: https://github.com/andreyz4k/StitchBindings.jl
- Created by: @andreyz4k
- Version: v0.1.0
- Commit: 72730b1ebe81f9927e5ec7e41756efba4fde6b3f
- Reviewed by: @andreyz4k
- Reference: https://github.com/andreyz4k/StitchBindings.jl/commit/72730b1ebe81f9927e5ec7e41756efba4fde6b3f#commitcomment-146189494
<!-- bf0c69308befbd3ccf2cc956ac8a46712550b79fc9bfb5e4edf8f833f05f4c18b06eddad8845b45beb9f45c2b8020dd622504d68c1cb3c085d1f89bb548565d242ca213b01213b67e47c9cd0d519fc7997fb1d26e56140fdeee3d8b159f249c498dff3cae46403a20e5bebb5dcad8b7ed326c6536eaedb0bba411bd90624dc99668a34955f82f92bf52760440a3fb7b0fdaf6a42bd1f2ec5058707dc403e81254e781df2ad52f249868541e0f07714b2fd4b8d47835821b5590a0af7964ccabea53316eb50bf603eef9c9320981f31373462af34c3c99ac9ab7870395e9ca25b -->

goerz commented:
Thank you for submitting your package! However, please make sure to add some documentation before registering. [At the very least](https://discourse.julialang.org/t/when-does-it-make-sense-to-post-a-new-package-to-the-julia-repository/108701/4), that would be a description of the package's purpose and a small usage example in the README

JuliaRegistrator committed:
New package: StitchBindings v0.1.0

UUID: 98c25cc8-eb50-4420-96bf-d3053e46f68d
Repo: https://github.com/andreyz4k/StitchBindings.jl.git
Tree: d21469ebcb7cb97b6ef689bdcaa30a4a2c2c807a

Registrator tree SHA: 17aec322677d9b81cdd6b9b9236b09a3f1374c6a

andreyz4k commented:
Well, that's the thing with working in a fairly unpopular research domain. The whole field of discrete program synthesis is maybe a couple of hundred active people and I might be the only one who is actively using Julia at this moment. On the other hand, Stitch is quite well-known in the field with 70 stars for the main repo and `stitch_bindings` library in pip with the Python bindings. So there is a potential for other people to use this bindings package but it's highly dependent on the results of my main research project and its ability to draw more people into the ecosystem.

Regarding the documentation, Stitch is a very specialized piece of software, so there is no way somebody will try to use this bindings library without knowing what it does, so I believe that there is no need to duplicate explanations for its functionality, and it's enough to explain the interface in the terms that are used in Stitch's documentation/paper. I agree that my current README is minimal, but it would be enough for me as a person who is familiar with Stitch. I might be wrong regarding other people, but I currently don't have an audience to check.

All in all, I would like to add better documentation, more tests, and so on, but right now it's not my top priority. The interface can change, and I would very much prefer to focus on more pressing parts of my project. It's version 0.1.0 after all.
[noblock]

goerz commented:
I get your point, but I would stress that there's basically never a good excuse not to write great documentation (it doesn't have to be large volumes, but it should be *good*, and address a wide audience, at some level). This is especially true when you say

> it's highly dependent on the results of my main research project and its ability to draw more people into the ecosystem.

I believe that the number one way to draw people into your research, or into your package or its surrounding ecosystem is to invest in documentation. You should not just be addressing the dozen or so people in the world that are in your exact niche and might be the most obvious users of your software.

> Stitch is a very specialized piece of software, so there is no way somebody will try to use this bindings library without knowing what it does

I wouldn't consider it inconceivable that I could be in your target audience at some level. I work in quantum optimal control theory / optimization, I've implemented a computer algebra system, and I have a general interest in computer science topics. It's absolutely possible the program synthesis could provide ideas for my work, and that I might want to play around a little with a package like `StichBindings`. Of course, I had to follow the link to [stitch](https://github.com/mlb2251/stitch) and then read through the linked [arXiv paper](https://arxiv.org/abs/2211.16605) to even get to the term "program synthesis" and a vague notion of what the package might be about. I believe that it is important to have a paragraph or two in a package README/documentation (or in the introduction of a paper), that gives an educated lay person a basic context for the kind of problem that is being solved. That's all I was asking for here, really: A paragraph at the level of someone with a science degree asking you at a party "So, what do you work on?"/"What's this software about that you're writing?".

> The whole field of discrete program synthesis is maybe a couple of hundred active people and I might be the only one who is actively using Julia at this moment.

With something that niche, I might push for a more niche name as well. "Stitch" is going to be one of those names that are very overloaded. Just type "Stitch" into the GitHub search. In that sense, `StichBindings` is very generic. Now, if a package comes along that's very polished, I'd tend to go with the "first-come-first-serve" approach. But otherwise, I would advocate for a more "descriptive" name, maybe something like `StitchSynthesisBindings`. With (long) names that don't really have the potential to clash with future registrations, at least my level of scrutiny drops dramatically. I would certainly not care enough to put any stumbling blocks in the way of `StitchSynthesisBindings`.

goerz commented:
I'm not going to "veto" this, though. I think I made the point for maybe using a longer name, but let me know if you want to continue with `StitchBindings`, and I'll unblock the registration.

Of course, I'd still encourage you to include a brief description of what the package, respectively the wrapped `stitch` does. But you can also do that later, while polishing the package.

[noblock]

andreyz4k commented:
I chose the name for consistency with the Python package `stitch_bindings` with the same functionality. And I understand that the whole "what does stitch do" can be confusing also because it implements a very cool algorithm that only makes sense as a part of a larger system and not as a standalone package. But I'll try to add a more detailed explanation later.

[noblock]
